Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java

```diff
@@ -2,13 +2,94 @@
 
 import androidx.appcompat.app.AppCompatActivity;
 
+import android.content.Context;
 import android.os.Bundle;
 
+import android.preference.PreferenceManager;
+
+import org.osmdroid.config.Configuration;
+import org.osmdroid.tileprovider.tilesource.TileSourceFactory;
+import org.osmdroid.views.MapView;
+
+
+
+import android.view.View;
+import android.widget.TextView;
+
+import com.teamapricot.projectwalking.handlers.CameraHandler;
+import com.teamapricot.projectwalking.photos.PhotoController;
+
+import com.teamapricot.projectwalking.Navigation.MapController;
+
 public class MainActivity extends AppCompatActivity {
+    private PhotoController photoController;
+    LocationHandler locationHandler;
+
+    MapView map = null;
+    private MapController mapController = null;
+
 
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
+    public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+
+        //handle permissions first, before map is created. not depicted here
+
+        //load/initialize the osmdroid configuration, this can be done
+        Context ctx = getApplicationContext();
+        Configuration.getInstance().load(ctx, PreferenceManager.getDefaultSharedPreferences(ctx));
+        //setting this before the layout is inflated is a good idea
+        //it 'should' ensure that the map has a writable location for the map cache, even without permissions
+        //if no tiles are displayed, you can try overriding the cache path using Configuration.getInstance().setCachePath
+        //see also StorageUtils
+        //note, the load method also sets the HTTP User Agent to your application's package name, abusing osm's tile servers will get you banned based on this string
+
+        //inflate and create the map
         setContentView(R.layout.activity_main);
+
+        map = (MapView) findViewById(R.id.map);
+        map.setTileSource(TileSourceFactory.MAPNIK);
+    }
+
+    @Override
+    public void onResume(){
+        super.onResume();
+        //this will refresh the osmdroid configuration on resuming.
+        //if you make changes to the configuration, use
+        //SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
+        //Configuration.getInstance().load(this, PreferenceManager.getDefaultSharedPreferences(this));
+        map.onResume(); //needed for compass, my location overlays, v6.0.0 and up
+    }
+
+    @Override
+    public void onPause(){
```

Comment:
This should be `protected` instead of `public`, since that is what the base class assumes. Otherwise it won't be overridden correctly. The base class in this case is simply the `Activity` class, see the link for details:

https://developer.android.com/reference/android/app/Activity.html#protected-methods

And here is the corresponding class inheritance:

https://developer.android.com/reference/androidx/appcompat/app/AppCompatActivity